exonum/src/runtime/rust/api.rs

```diff
@@ -116,6 +116,19 @@ pub fn endpoints(runtime: &RustRuntime) -> impl IntoIterator<Item = (String, Api
         })
         .collect::<HashMap<_, _>>();
     let exonum_sources = exonum_proto_sources();
+    // Cache filtered sources to avoid expensive operations in the endpoint handler.
+    let filtered_sources = artifact_proto_sources
+        .clone()
```

Comment:
Cloning seems to be unnecessary (unless I'm missing something, `artifact_proto_sources` isn't used anywhere else). Thus, it could make sense to combine creation of `filtered_sources` with creation of `artifact_proto_sources`.

Reply:
Works without `clone()`.